components/infobox/wikis/mobilelegends/infobox_person_player_custom.lua

```diff
@@ -115,18 +163,65 @@ function CustomPlayer:createWidgetInjector()
 end
 
 function CustomPlayer:adjustLPDB(lpdbData)
-	lpdbData.extradata.isplayer = _role.isPlayer or 'true'
-	lpdbData.extradata.role = _role.role
-	lpdbData.extradata.role2 = _role2.role
+	lpdbData.extradata.role = Variables.varDefault('role')
+	lpdbData.extradata.role2 = Variables.varDefault('role2')
+	lpdbData.extradata.role3 = Variables.varDefault('role3')
 
-	-- store signature heroes with standardized name
-	for heroIndex, hero in ipairs(Player:getAllArgsForBase(_args, 'hero')) do
-		lpdbData.extradata['signatureHero' .. heroIndex] = HeroNames[hero:lower()]
-	end
+	lpdbData.extradata.hero1 = _args.hero1 or _args.hero
+	lpdbData.extradata.hero2 = _args.hero2
+	lpdbData.extradata.hero3 = _args.hero3
+	lpdbData.extradata.hero4 = _args.hero4
+	lpdbData.extradata.hero5 = _args.hero5
+	lpdbData.type = CustomPlayer._isPlayerOrStaff()
+
+	lpdbData.region = Template.safeExpand(mw.getCurrentFrame(), 'Player region', {_args.country})
```

Comment:
i am against using shitty template expand instead of using the semi standardized region module!